### DISCUSSION TITLE
How to get a API Key for Homepage integration

### DISCUSSION THREAD

**Discussion by hunkyn** [MDeLuise/plant-it#176] [Q&A] [answered]
Hello, 

I am following the documentation to get an API Key to integrate with homepage and based on that I went to this url

http://192.168.30.9:9080/api/swagger-ui/index.html 

and using the option to get a new key but i keep getting a 403 error. I understand i have to authorize but what is the password for that? Is that the password i use to login or the password of the database or something else?

**Comment by MDeLuise** [comment, marked as answer]
Hi,
you first need to authenticate in order to create the API Key.
This is the all process:
* go to `http://192.168.30.9:9080/api/swagger-ui/index.html`
* get the jwt token calling the `POST /authentication/login` with your credentials
* set the jwt token in swagger for the next calls (use the "authorize" button at the top)
* create the new api key calling `POST /api-key` optionally passing a name for the key

I realize that it would be useful to have the possibility of managing the API key directly from the app. I will implement this feature soon.

**Comment by hunkyn** [comment]
Got it and I was able to setup. Is it possible to add it to the documentation in the meantime? Happy to help update the documentation as well if needed.

**Reply by MDeLuise** [reply]
> Is it possible to add it to the documentation in the meantime?

Of course, thanks for pointing this out.
I updated the documentation' sources in order to include this info (see the screenshot below).
The online documentation will be updated soon.

<img width="1217" alt="Screenshot 2024-05-27 at 18 05 39" src="https://github.com/MDeLuise/plant-it/assets/66636702/0d9c2e9c-f7bc-4926-a273-fac1bd2c7f43">
